src/python/WMComponent/TaskArchiver/DataCache.py

```diff
@@ -0,0 +1,12 @@
+import time
+
+class DataCache(object): 
+    _finishedWFs = [] # Global cache holding workflow lists 
+    
+    @staticmethod
+    def getFinishedWorkflows():
+        return DataCache._finishedWFs;
+
+    @staticmethod
+    def setFinishedWorkflows(workflowLists):
+        DataCache._finishedWFs = workflowLists;
```

Comment:
cache need to be overwritten each polling cycle.
https://github.com/dmwm/WMCore/pull/6585/files#diff-748e5dcfaf8a48d0e9774b3066780e42R298

I am passing set for the value, sorry it my mistake set the default value as list. I will fix that.